### DISCUSSION TITLE
Check IP, settings Payouts

### DISCUSSION THREAD

**Discussion by belmix** [coinfoundry/miningcore#930] [General]
I did not find in the code checking ip
![image](https://user-images.githubusercontent.com/4902701/133067783-84819163-8a63-4fb3-a83f-2b32dcbda6d2.png)
![image](https://user-images.githubusercontent.com/4902701/133067819-6f08798c-1bf9-4924-b123-64047da0ca2a.png)
the table also does not contain the miner's ip
if you look at the code, any user can change the payment settings for any account
![image](https://user-images.githubusercontent.com/4902701/133067883-a9396015-9d8f-415e-88f1-c2ff98b99ab1.png)
The IP address is only in the table of shares, but it is not related to the validation of the request sent.

**Comment by oliverw** [comment]
> The IP address is only in the table of shares, but it is not related to the validation of the request sent.

Wrong. The latter part is the whole point of the implementation.

**Comment by belmix** [comment]
```
{"paymentThreshold":"0",
"PoolId":"dgb",
"Address":"DGoajbBjSSqqVhDq5Hk6dQxh8CLnZFL2ir"}
```
ip address should also be in the request?

**Reply by oliverw** [reply]
Of course.

**Comment by belmix** [comment]
```
{"paymentThreshold":"0",
"PoolId":"dgb",
"Address":"DGoajbBjSSqqVhDq5Hk6dQxh8CLnZFL2ir",
"ip":"x.x.x.x"}

```
right?

**Comment by oliverw** [comment]
My apologies. I didn't realize that `master` still has the old implementation. The current implementation is in the dev branch (`oliverw`): https://github.com/coinfoundry/miningcore/blob/oliverw/src/Miningcore/Api/Controllers/PoolApiController.cs

I will push a new release later today.

**Comment by belmix** [comment]
ok thx, 
I will add FCH and DVT this week
I don't know, maybe anyone needs it, I can still unload the customs fee
https://github.com/belmix/MinigcoreNET5/commits/master
![image](https://user-images.githubusercontent.com/4902701/133082219-3ea2984b-593d-486f-9680-fae72457c946.png)
implementation allows you to make not only individual fee, but can also be used as a blacklist
"percentageFeeCustom" : 1.0,

**Comment by riksarchen** [comment]
Please add description how to use this for the average user

**Reply by belmix** [reply]
![923CD31B-B2C4-4DBD-B40F-44214EAC54F0](https://user-images.githubusercontent.com/4902701/133684554-8d744beb-a95b-4d55-a0a0-0637e22bec42.jpeg)
is it relevant or is the structure different?

**Reply by oliverw** [reply]
> ![923CD31B-B2C4-4DBD-B40F-44214EAC54F0](https://user-images.githubusercontent.com/4902701/133684554-8d744beb-a95b-4d55-a0a0-0637e22bec42.jpeg)
> is it relevant or is the structure different?

`paymentthreshold` does not contain an underscore.

**Reply by belmix** [reply]
Example for DVT Coin:

SET ROLE miningcore;
DROP TABLE miner_settings;
CREATE TABLE miner_settings
(
 poolid TEXT NOT NULL PRIMARY KEY,
 address TEXT NOT NULL,
 paymentthreshold decimal(28,12) NOT NULL,
 created TIMESTAMP NOT NULL,
 updated TIMESTAMP NOT NULL
) PARTITION BY LIST (poolid);

CREATE TABLE miner_settings_dvt PARTITION OF miner_settings FOR VALUES IN ('dvt');

**Comment by belmix** [comment]
👍👍👍👍👍
